Library/Homebrew/rubocops/lines.rb

```diff
@@ -157,6 +157,23 @@ def unless_modifier?(node)
         end
       end
 
+      class MpiChoice < FormulaCop
+        def audit_formula(node, _class_node, _parent_class_node, body_node)
+          return unless formula_tap == "homebrew-core"
+
+          find_method_with_args(body_node, :depends_on, "mpich") do |source_node|
+            problem "'depends_on \"mpich\"' should be replaced with 'depends_on \"open-mpi\"'."
+            offending_node(source_node)
```

Comment:
I tried many different iterations of this and I cannot seem to get any diagnostic info in the corresponding test.

Reply:
@zbeekman What are you trying to do, what have you tried and what did you expect to happen?

May be worth pulling the cop changes into a separate PR?

Reply:
As far as I can tell, the functionality for the cop is working as expected. I just don't understand how to write the test for it correctly. I could factor out the test into a new PR. I've tested the audit/style cop extensively locally, and it functions as I would expect.

Reply:
Sorry but don't want to add a new cop without the test. Can you extract them both into a new PR?

Reply:
PR #6214 is open to debug